includes/class-edd-html-elements.php

```diff
@@ -524,28 +527,31 @@ public function year_dropdown( $name = 'year', $selected = 0, $years_before = 5,
 	 *
 	 * @since 1.5.2
 	 *
-	 * @param string $name     Name attribute of the dropdown
-	 * @param int    $selected Month to select automatically
-	 *
+	 * @param string $name     Name attribute of the dropdown.
+	 * @param int    $selected Month to select automatically.
+	 * @param string $id A unique identifier for the field.
 	 * @return string $output Month dropdown
 	 */
-	public function month_dropdown( $name = 'month', $selected = 0 ) {
+	public function month_dropdown( $name = 'month', $selected = 0, $id = 'edd-month-dropdown' ) {
 		$month    = 1;
 		$options  = array();
 		$selected = empty( $selected ) ? date( 'n' ) : $selected;
 
 		while ( $month <= 12 ) {
-			$options[ absint( $month ) ] = edd_month_num_to_name( $month );
+			$options[ absint( $month ) ] = edd_month_num_to_name( $month, true );
```

Comment:
Should the `true` parameter be added as a parameter to this method as well after the new `$id` parameter? So that anyone who has been using the `month_dropdown` method expecting it to output three letter month names will still have that?

Reply:
I have mixed feelings on this, since the primary function doesn't use that parameter.  Is there a way I could add it in the `while` loop?

Something more like this but done in the proper PHP way?

```
while ( $month <= 12 ) {
        @since 3.0
        @param bool $return_long_name Optional. Return full name of month if true. Default false.
        $options[ absint( $month ) ] = edd_month_num_to_name( $month, true );
	$month ++;
}
```

Reply:
Sorry, I think I asked my question unclearly. I don't see it used in EDD core outside of the Exports screen, but I'm thinking we should update the month_dropdown parameters to include the `$return_long_name` parameter that you've already added to the `edd_month_num_to_name` function, and then use it in the exports screen with the new parameter to get the long name. So the method would be updated like this:

```php
	/**
	 * Renders an HTML Dropdown of months
	 *
	 * @since 1.5.2
	 *
	 * @param string  $name             Name attribute of the dropdown.
	 * @param int     $selected         Month to select automatically.
	 * @param string  $id               A unique identifier for the field.
	 * @param boolean $return_long_name Whether to use the long name for the month.
	 *
	 * @return string $output Month dropdown
	 */
	public function month_dropdown( $name = 'month', $selected = 0, $id = 'edd_month_select', $return_long_name = false ) {
		$month    = 1;
		$options  = array();
		$selected = empty( $selected ) ? date( 'n' ) : $selected;

		while ( $month <= 12 ) {
			$options[ absint( $month ) ] = edd_month_num_to_name( $month, $return_long_name );
			$month ++;
		}
...
```

And we would use it:

```php
echo EDD()->html->month_dropdown( 'start_month', 0, 'edd_export_earnings', true );
```

That way we won't be making unexpected changes to any extensions using the `month_dropdown` method ([ref](https://github.com/search?q=org%3Aeasydigitaldownloads+html-%3Emonth_dropdown&type=code)).

Reply:
That makes way more sense!  :joy: